src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,125 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface NewMovieProps {
+  onAdd: (movie: Movie) => void,
+}
+
+export const NewMovie = ({ onAdd }: NewMovieProps) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const defaultMovie = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState<Movie>({ ...defaultMovie });
+
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbId,
+    imdbUrl,
+  } = movie;
+
+  const isValid = title.length && imgUrl.length
```

Comment:
Empty string is always falsy itself
```suggestion
  const isValid = title&& imgUrl
```